sdk/data/azcosmos/cosmos_client_test.go

```diff
@@ -102,8 +100,8 @@ func TestEnsureErrorIsGeneratedOnResponse(t *testing.T) {
 	}
 
 	asError := err.(*azcore.ResponseError)
-	if asError.ErrorCode != someError.Code {
-		t.Errorf("Expected %v, but got %v", someError.Code, asError.ErrorCode)
+	if asError.ErrorCode != "404 Not Found" {
```

Comment:
With the new error, is the "SomeCode" body logged anywhere? This test should validate that the "SomeCode" is part of the error, maybe the body?